src/main/java/emissary/core/IBaseDataObjectHelper.java

```diff
@@ -260,4 +262,25 @@ public static byte[] findPreferredData(final IBaseDataObject payload, List<Strin
 
         return payload.data();
     }
+
+    /**
+     * Get the effective parameters for a given payload, considering parameters inherited from the parents
+     * 
+     * @param payload Get parameters from this IBaseDataObject
+     * @return The full set parameters including inherited ones
+     */
+    public static Map<String, Collection<Object>> getParametersWithInheritance(IBaseDataObject payload) {
+        IBaseDataObject parent = payload.getParent();
+        Set<String> inheritedParameters = payload.getInheritedParameters();
+        Map<String, Collection<Object>> effectiveParams = new HashMap<>(payload.getParameters());
+        if (parent != null) {
+            for (String key : parent.getParameterKeys()) {
+                if (inheritedParameters == null || inheritedParameters.contains(key)) {
```

Comment:
Only add the TLD parameter if current `BaseDataObject` parameter's doesn't contain the key.

Reply:
See LogFilter.java